### PR TITLE
Review practice

### PR DESCRIPTION
**Pull request by rcqx**
## Updates:
* A method was utilized to deleted selected items from list (clearList());
* While using clearList() method, index numbers get updated to actual position values 
* Update items object's value for completed key upon user actions.

image

### CODE REVIEW

File: src/modules/todosListClass.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          this.UpdateList();
+          const task = document.getElementById(`task${i}`);
+          task.style.textDecoration = 'line-through';
+        } else if (this.List[i].completed === true) {
+          this.List[i].completed = false;
+          const stringData = JSON.stringify(this.List);
+          localStorage.setItem('todoList', stringData);
+          this.UpdateList();
+          const task = document.getElementById(`task${i}`);
+          task.style.textDecoration = 'none';
```
Comment (acolombo1) [inline src/modules/todosListClass.js:116]: Please use a name that reflects what the method does (removeItemFromLocalStorage)

File: src/modules/todosListClass.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        }
+      });
+    }
+  }
+
+  clearList() {
+    document.getElementById('clearA').addEventListener('click', () => {
```
Comment (acolombo1) [inline src/modules/todosListClass.js]: Again, let's change clearA for a name that has a more relevant meaning, wouldn't be nice?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// hide remove icon from item on mouseout
+todos.hideRemoveIcon();
+
+// Remove items when clicking X icon
+todos.removeItem();
+
+// update task
+todos.updateTask();
+
```
Comment (acolombo1) [inline src/index.js]: Would you please clarify why you need to use all these methods upon windows load? You display RemoveIcon then you hide RemoveIcon, for example, that confuses me.

File: src/modules/todosListClass.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      for (let i = 0; i < this.List.length; i += 1) {
+        const listedItem = document.getElementById(`item${i}`);
+        const filtered = this.List.filter((items) => items.completed === false);
+        const stringData = JSON.stringify(filtered);
+        localStorage.setItem('todoList', stringData);
+        listedItem.remove();
+        this.UpdateList();
+        this.updateIndex();
+        window.location.reload();
```
Comment (acolombo1) [inline src/modules/todosListClass.js]: Kindly avoid windows reload if possible in your code, it means an HTTP request and waiting for a response from the server.

**Comment by acolombo1** [comment]
Great job, I see your app is fully functional without glitches. The review in this case focused on code best practices to be able to get better at that. Please don't forget that we can comment further on the issues, thank you!

**Review by acolombo1 — CHANGES_REQUESTED** [review]
Hi @jr-cast,
Great job with your code, I see the application is fully functional and without glitches, the coding style has been scrutinized in this case, to give the opportunity to get better at that.

There are some issues that you still need to work on to go to the next project but you are almost there!

Highlights
✔️ No linter errors.
✔️ GitHub flow was correctly used.
✔️ Professional README.
✔️ Descriptive PR.

Required Changes ♻️
Check the comments under the review.

Cheers and Happy coding!👏👏👏

Thanks, and don't forget that we can comment further on the issues. Please, remember to tag me in your question so I can receive the notification @acolombo1.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/index.js (hunk 3 above)
Comment (rcqx) [reply src/index.js]: Hi Aldo, yes, that particular line (37) was redundant as I was already using it previously. Thanks for the comment. In regards to your question, I had to call the methods upon the todo list in the main.js as I declared them in the todo list class but I was not actually using them. If I removed those lines from my main.js file, the functionalities, although already existing in the class as methods, are not triggered and thus the page is not able to create, update or delete items from the local storage.

Comment (acolombo1) [reply src/index.js]: Oh, I see those are the addEventListener calls, the way they are named suggested to me that they were direct actions. Maybe you can find better namings next time, or maybe it's me with my first code review that I'm not skillful enough. Cheers!🤷‍♂️

File: src/modules/todosListClass.js (hunk 1 above)
Comment (rcqx) [reply src/modules/todosListClass.js:116]: Thanks, sometimes I start with one idea in mind then it changes and I forgot to revisit the actual name of the function! 🏎️

**Review by acolombo1 — COMMENTED** [review]
Hi @jr-cast ,

Your project is complete! There is nothing else to say other than... it's time to merge it ✈
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

**Pull request merged**